### DISCUSSION TITLE
Send keystrokes

### DISCUSSION THREAD

**Discussion by tonyrumans** [sezanzeb/input-remapper#423] [General]
Does this software allow you to send keystrokes by hitting a key combination? Like "Ctrl + Space" to send a password to whatever window is active? I thought that "key(w)" would send the letter "w" but it's not doing it. I have tried following the Macros documentation but I'm lost

**Comment by jonasBoss** [comment]
can you try "key(KEY_W)"? 
It is totally possible to send passwords or whatever key sequence you like. 

EDIT: your problem is most likely the use of "Ctrl+..." as an input combination:
[usage.md](https://github.com/sezanzeb/input-remapper/blob/main/readme/usage.md#combinations):
> Combinations involving Modifiers might not work. Configuring a combination of two keys to output a single key will require you to push down the first key, which of course ends up injecting that first key. Then the second key will trigger the mapping, because the combination is complete. This is not a bug. Otherwise every combination would have to automatically disable all keys that are involved in it.

There are a few workarounds for this in the usage.md documented. 
The `beta` branch has a proper solution to this.

**Reply by tonyrumans** [reply]
Well I think I see what might be going on. My keyboard combo is Ctrl and Space at the same time. In a text editor when I press those keys it recognizes the Ctrl key being pushed and it does send the "s" but doesn't write it to the editor. It thinks I'm pressing Ctrl S which is a Save command

**Comment by darkestmon** [comment]
Hi, are there easier means to send a series of characters like the author was trying to do?
for example, key(H).key(e).key(l).key(l).key(o) does the job but You still need to shift up the H to make it uppercase. Isn't there something like: type('Hello')?

Thanks!

**Reply by jonasBoss** [reply]
not really, you can shift the H to uppercase with the `modify` macro:
`modify(Shift_L, key(h)).key(e).key(l).key(l).key(o)`
